velox/type/Type.h

```diff
@@ -2009,6 +2009,34 @@ bool registerCustomType(
     const std::string& name,
     std::unique_ptr<const CustomTypeFactories> factories);
 
+// See registerOpaqueType() for documentation on type index and opaque type
+// alias.
+std::unordered_map<std::string, std::type_index>& getTypeIndexByOpaqueAlias();
+
+// Reverse of getTypeIndexByOpaqueAlias() when we need to look up the opaque
+// alias by its type index.
+std::unordered_map<std::type_index, std::string>& getOpaqueAliasByTypeIndex();
+
+std::type_index getTypeIdForOpaqueTypeAlias(const std::string& name);
+
+std::string getOpaqueAliasForTypeId(std::type_index typeIndex);
+
+// OpaqueType represents a type that is transparent to the Velox type system.
+// To identify the underlying type we use std::type_index which is stable
+// within the same process. However, it is not necessarily stable across
+// processes.
+//
+// So if we were to serialize an opaque type using its std::type_index, we
+// might not be able to deserialize it in another process. To solve this
+// problem, we require that both the serializing and deserializing processes
+// register the opaque type using registerOpaqueType() with the same alias.
```

Comment:
thanks for the comment. We usually use "///" for documentation header comments such as this.